.github/workflows/publish.yml

```diff
@@ -0,0 +1,35 @@
+on:
+  workflow_dispatch:
+  push:
+    branches: main
+
+name: Quarto Publish
+
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Set up Quarto
+        uses: quarto-dev/quarto-actions/setup@v2
+        
+      - name: Install R
+        uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: '4.4.0'
+          
+      - name: Install R Dependencies
+        uses: r-lib/actions/setup-renv@v2
+        with:
+          cache-version: 1
+
+      - name: Render and Publish
+        uses: quarto-dev/quarto-actions/publish@v2
+        with:
+          target: gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This action seems to force deploying a page via `gh-pages` branch, which isn't super desirable in my mind.

From https://github.com/quarto-dev/quarto-actions/tree/2b90fd62e27d260d01f63fff6364baa7c2f8d96f/publish#github-pages:

> Quarto needs to configure the repository for publishing through GitHub Actions. To do this, run quarto publish gh-pages locally, once. This will create a new branch called gh-pages and push it to the remote repository, and configure the gh-pages branch to be the [publishing source for GitHub Pages](https://quarto.org/docs/publishing/github-pages.html#source-branch).

In my mind [publishing using GitHub Actions is far preferable](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#publishing-with-a-custom-github-actions-workflow).

My suggestion is to break this into two separate jobs: a `build` job and a `deploy` job.

So something like https://github.com/scikit-hep/pyhf/blob/89a6c132c7edcf934eac52203b08945eadc4ad13/.github/workflows/docs.yml, which is what my suggested changes to use `quarto-dev/quarto-actions/render@v2` entail.

```suggestion
      - name: Render Quarto Project
        uses: quarto-dev/quarto-actions/render@v2
        with:
          to: html

    - name: Fix permissions if needed
      run: |
        chmod -c -R +rX "html/" | while read line; do
          echo "::warning title=Invalid file permissions automatically fixed::$line"
        done

    - name: Upload artifact
      uses: actions/upload-pages-artifact@v3
      with:
        path: 'html/'

  deploy:
    name: Deploy docs to GitHub Pages
    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
    needs: build
    # Set permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
    permissions:
      contents: read
      pages: write
      id-token: write

    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}

    runs-on: ubuntu-latest

    steps:
    - name: Setup Pages
      uses: actions/configure-pages@v5

    - name: Deploy to GitHub Pages
      id: deployment
      uses: actions/deploy-pages@v4
```

I don't know exactly how `https://github.com/quarto-dev/quarto-actions/tree/main/render` works in terms of creating directory output, so you might want to check that before using `html/`.

The advantage of splitting out build and deploy is that you can fully test the build and then deploy as needed.

This also currently only deploy on push events  to `main`, but maybe you want something different.